spy/ast_dump.py

```diff
@@ -76,8 +87,15 @@ def is_complex(obj: Any) -> bool:
         multiline = any(is_complex_field)
         #
         if node is self.highlight:
-            color = 'red'
-        self.write(name, color=color)
+            text_color = 'red'
+        # If the vm contains an expr_color_map, use the expression's color
+        # as the text background color
+        bg_color = None
+        if self.vm and self.vm.expr_color_map:
+            color: Color | None = self.vm.expr_color_map.get(node, None)
```

Comment:
+0.5 to pass around `vm` if it helps, but note that we could also pass just `expr_color_map`, and cut the circular dependency between `ast.py` and `vm.py.`
Anyway, not a big deal.

Reply:
Yeah, I decided to pass in `vm` since as more features get added, I didn't want to pass more and more arguments to the various pretty print functions. Then again, we probably don't want `vm` to become a "dumping ground" of global variables. But at least for now, passing in `vm` seemed like a good compromise.